src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/DbReference.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.workbench.leonardo.db
+
+import java.sql.SQLTimeoutException
+
+import akka.actor.ActorSystem
+import com.google.common.base.Throwables
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import liquibase.{Contexts, Liquibase}
+import liquibase.database.jvm.JdbcConnection
+import liquibase.resource.{ClassLoaderResourceAccessor, ResourceAccessor}
+import org.broadinstitute.dsde.workbench.leonardo.config.LiquibaseConfig
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIO
+import slick.jdbc.{JdbcBackend, JdbcDataSource, JdbcProfile, TransactionIsolation}
+import net.ceedubs.ficus.Ficus._
+import sun.security.provider.certpath.SunCertPathBuilderException
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object DbReference extends LazyLogging {
+
+  private def initWithLiquibase(dataSource: JdbcDataSource, liquibaseConfig: LiquibaseConfig, changelogParameters: Map[String, AnyRef] = Map.empty): Unit = {
+    val dbConnection = dataSource.createConnection()
+    try {
+      val liquibaseConnection = new JdbcConnection(dbConnection)
+      val resourceAccessor: ResourceAccessor = new ClassLoaderResourceAccessor()
+      val liquibase = new Liquibase(liquibaseConfig.changelog, resourceAccessor, liquibaseConnection)
+
+      changelogParameters.foreach { case (key, value) => liquibase.setChangeLogParameter(key, value) }
+      liquibase.update(new Contexts())
+    } catch {
+      case e: SQLTimeoutException =>
+        val isCertProblem = Throwables.getRootCause(e).isInstanceOf[SunCertPathBuilderException]
+        if (isCertProblem) {
+          val k = "javax.net.ssl.keyStore"
+          if (System.getProperty(k) == null) {
+            logger.warn("************")
+            logger.warn(s"The system property '${k}' is null. This is likely the cause of the database connection failure.")
+            logger.warn("************")
+          }
+        }
+        throw e
+    } finally {
+      dbConnection.close()
+    }
+  }
+
+  def init(config: Config, system: ActorSystem)(implicit executionContext: ExecutionContext): DbReference = {
+    val db = "mysql"
+    val dbConfig = DatabaseConfig.forConfig[JdbcProfile](db, config)
+
+    system.registerOnTermination {
+      dbConfig.db.shutdown
+    }
+
+    val liquibaseConf = config.as[LiquibaseConfig]("liquibase")
+    if (liquibaseConf.initWithLiquibase)
+      initWithLiquibase(dbConfig.db.source, liquibaseConf)
+
+    DbReference(dbConfig)
+  }
+}
+
+case class DbReference(private val dbConfig: DatabaseConfig[JdbcProfile])(implicit val executionContext: ExecutionContext) extends AllComponents {
```

Comment:
I think it's definitely worthwhile. The reason: slick calls are blocking. If there are a lot of expensive DB queries happening, we want the application to still be able to serve requests (example: the proxy endpoint might not require DB access). The best way to do that is to have separate thread pools for DB operations and serving HTTP requests.

Now we could either define a `fixedThreadPool` in Scala like Rawls, or define one in the Akka reference.conf. Maybe we should try the latter. So, in `reference.conf`:
```
slick-dispatcher {
  type = Dispatcher
  executor = "thread-pool-executor"
  thread-pool-executor {
    fixed-pool-size = 32
  }
  throughput = 1
}
```
And in `Boot.scala`:
``` 
val databaseExecutionContext: ExecutionContext = context.system.dispatchers.lookup("slick-dispatcher")
val dbRef = DbReference.init(config, system)(databaseExecutionContext)
```
Docs:
http://doc.akka.io/docs/akka/current/scala/dispatchers.html
http://doc.akka.io/docs/akka/current/scala/dispatchers.html#solution-dedicated-dispatcher-for-blocking-operations